lib/SIL/SILArgument.cpp

```diff
@@ -26,12 +26,6 @@ using namespace swift;
 SILArgument::SILArgument(SILBasicBlock *ParentBB, SILType Ty,
                          const ValueDecl *D)
   : ValueBase(ValueKind::SILArgument, Ty), ParentBB(ParentBB), Decl(D) {
-  // Function arguments need to have a decl.
```

Comment:
I'm not a fan of this particular assertion, but you might want to check with @adrian-prantl to make sure the debugger's able to do something reasonable with the self metadata.